src/Microsoft.ML.KMeansClustering/KMeansCatalog.cs

```diff
@@ -17,19 +18,41 @@ public static class KMeansClusteringExtensions
         /// Train a KMeans++ clustering algorithm.
         /// </summary>
         /// <param name="ctx">The regression context trainer object.</param>
-        /// <param name="features">The features, or independent variables.</param>
+        /// <param name="featureColumn">The features, or independent variables.</param>
         /// <param name="weights">The optional example weights.</param>
         /// <param name="clustersCount">The number of clusters to use for KMeans.</param>
-        /// <param name="advancedSettings">Algorithm advanced settings.</param>
+        /// <example>
+        /// <format type="text/markdown">
+        /// <![CDATA[
+        ///  [!code-csharp[SDCA](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/KMeans_example.cs)]
+        /// ]]></format>
+        /// </example>
         public static KMeansPlusPlusTrainer KMeans(this ClusteringContext.ClusteringTrainers ctx,
-           string features,
+           string featureColumn = DefaultColumnNames.Features,
            string weights = null,
-           int clustersCount = KMeansPlusPlusTrainer.Defaults.K,
-           Action<KMeansPlusPlusTrainer.Arguments> advancedSettings = null)
+           int clustersCount = KMeansPlusPlusTrainer.Defaults.K)
         {
             Contracts.CheckValue(ctx, nameof(ctx));
             var env = CatalogUtils.GetEnvironment(ctx);
-            return new KMeansPlusPlusTrainer(env, features, clustersCount, weights, advancedSettings);
+            var options = new KMeansPlusPlusTrainer.Options
+            {
+                FeatureColumn = featureColumn,
+                WeightColumn = weights != null ? Optional<string>.Explicit(weights) : Optional<string>.Implicit(DefaultColumnNames.Weight),
+                K = clustersCount
+            };
+            return new KMeansPlusPlusTrainer(env, options);
+        }
+
+        /// <summary>
+        /// Train a KMeans++ clustering algorithm.
+        /// </summary>
+        /// <param name="ctx">The regression context trainer object.</param>
+        /// <param name="options">Algorithm advanced settings.</param>
+        public static KMeansPlusPlusTrainer KMeans(this ClusteringContext.ClusteringTrainers ctx, KMeansPlusPlusTrainer.Options options = null)
+        {
+            Contracts.CheckValue(ctx, nameof(ctx));
+            var env = CatalogUtils.GetEnvironment(ctx);
+            return new KMeansPlusPlusTrainer(env, options);
```

Comment:
>options [](start = 50, length = 7)

Host.CheckValue(options, nameof(options));
You have this in `internal KMeansPlusPlusTrainer(IHostEnvironment env, Options options)`
if no one will specify options that check will throw.
Is it works? Do we somehow magically create Options somewhere?